test/odbc/test_pg_datatype.cpp

```diff
@@ -0,0 +1,160 @@
+#include <gtest/gtest.h>
+#include <sqlext.h>
+#include "odbc_handler.h"
+#include "query_generator.h"
+
+using std::pair;
+
+class PSQL_DataTypes : public testing::Test{
+
+};
+
+TEST_F(PSQL_DataTypes, Datatypes_bigint) {
+
+  const string TABLE_NAME = "master_dbo.bigint_table";
+  const string COL1_NAME = "pk"; // primary key
+  const string COL2_NAME = "data";
+  const string DATATYPE_NAME = "sys.bigint";
+
+  const int BYTES_EXPECTED = 8;
+  const int NUM_ROWS = 4;
+  const int LENGTH_EXPECTED = 20;
+  const int PRECISION_EXPECTED = 0;
+  const int SCALE_EXPECTED = 0;
+  const int BUFFER_SIZE = 256;
+  const string NAME_EXPECTED = "int8";
+  
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  long long int pk;
+  long long int data;
+  int length;
+  int precision;
+  int scale;
+  char name[BUFFER_SIZE];
+  int strlength;
+
+  vector<pair<string, string>> BIGINT_TABLE_COLUMNS = {
+    {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME}
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns = {
```

Comment:
Sorry, it's been a while since I looked at ODBC calls. What does this do again?

Reply:
it's to set up the vector for the BindColumn function in `odbc_handler.cpp` which is just a helper to call SQLBindCol. It's bIasically the set of parameters we want to use when we call SQLBindCol to map variables to row results (this way, we don't have to call SQLBindCol  `n` times if we want to bind variables for a table with `n `columns).